test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -828,6 +829,142 @@ def init_proc_mon(self):
+                                                    pbs_conf_filepath)
+            if pbs_conf_val is None or len(pbs_conf_val) < 1:
+                raise ValueError("Could not parse pbs.conf on host %s" %
+                                 (server.hostname))
+
+            # Set all start bits
+            if (pbs_conf_val["PBS_START_SERVER"] != "1"):
+                pbs_conf_val["PBS_START_SERVER"] = "1"
+                dmns_to_start += 1
+                cmds_to_exec.append(["server", "start"])
+            if (pbs_conf_val["PBS_START_SCHED"] != "1"):
+                pbs_conf_val["PBS_START_SCHED"] = "1"
+                cmds_to_exec.append(["sched", "start"])
+                dmns_to_start += 1
+            if self.moms is None or len(self.moms) == 0:
+                if (pbs_conf_val["PBS_START_MOM"] != "1"):
+                    pbs_conf_val["PBS_START_MOM"] = "1"
+                    cmds_to_exec.append(["mom", "start"])
+                    dmns_to_start += 1
+            else:
+                pbs_conf_val["PBS_START_MOM"] = "0"
+            if self.comms is None or len(self.comms) == 0:
+                if (pbs_conf_val["PBS_START_COMM"] != "1"):
+                    pbs_conf_val["PBS_START_COMM"] = "1"
+                    cmds_to_exec.append(["comm", "start"])
+                    dmns_to_start += 1
+            else:
+                pbs_conf_val["PBS_START_COMM"] = "0"
+                cmds_to_exec.append(["comm", "stop"])
+
+            # Set PBS_CORE_LIMIT, PBS_SCP and PBS_SERVER
+            pbs_conf_val["PBS_CORE_LIMIT"] = "unlimited"
+            pbs_conf_val["PBS_SERVER"] = primary_server.hostname
+            if "PBS_SCP" not in pbs_conf_val:
+                scppath = self.du.which(server.hostname, "scp")
+                if scppath != "scp":
+                    pbs_conf_val["PBS_SCP"] = scppath
+
+            self.du.set_pbs_config(server.hostname, pbs_conf_filepath,
+                                   confs=pbs_conf_val)
+
+            if dmns_to_start == 4:
+                # Restart all
+                server.restart(server.hostname)
+            else:
+                for initcmd in cmds_to_exec:
+                    # start/stop the particular daemon
+                    server.pi.initd(server.hostname, initcmd[1],
+                                    pbs_conf_filepath, daemon=initcmd[0])
+                # Send a HUP to all daemons
+                self.server.signal("-HUP")
+                self.mom.signal("-HUP")
+                self.comm.signal("-HUP")
+                for sched in self.scheds.values():
+                    sched.signal("-HUP")
+
+        # Revert mom hosts' pbs.conf
+        for mom in self.moms.values():
+            start_mom = False
+            pbs_conf_filepath = self.du.get_pbs_conf_file(mom.hostname)
+            pbs_conf_val = self.du.parse_pbs_config(mom.hostname,
+                                                    pbs_conf_filepath)
```

Comment:
By doing this, you will keep any pbs.conf variables that are set during a test that are not the default values (e.g. PBS_LOG_HIGHRES_TIMESTAMP).  You need to create a dictionary of the variables you want, and set them.

Reply:
hmm .. ok, this will need some refactoring, so I'll mark this WIP